test/ecto/adapters/myxql_test.exs

```diff
@@ -230,6 +230,44 @@ defmodule Ecto.Adapters.MyXQLTest do
       ~s{INNER JOIN `target_rows` AS t1 ON t1.`id` = s0.`id`}
   end
 
+  test "parent binding subquery and CTE" do
+    initial_query =
+      "categories"
+      |> where([c], c.id == parent_as(:parent_category).id)
+      |> select([:id, :parent_id])
+
+    iteration_query =
+      "categories"
+      |> join(:inner, [c], t in "tree", on: t.parent_id == c.id)
+      |> select([:id, :parent_id])
+
+    cte_query = initial_query |> union_all(^iteration_query)
+    
+    breadcrumbs_query =
+      "tree"
+      |> recursive_ctes(true)
+      |> with_cte("tree", as: ^cte_query)
+      |> select([t], %{breadcrumbs: fragment("GROUP_CONCAT(? SEPARATOR ' / ')", t.id)})
+
+    query =
+      from(c in "categories",
+        as: :parent_category,
+        left_lateral_join: b in subquery(breadcrumbs_query),
```

Comment:
This test is currently failing because the MyXQL adapter doesn't handle lateral joins yet. [Seems like lateral joins were added in MySQL v8.0.14](https://dev.mysql.com/doc/refman/8.0/en/lateral-derived-tables.html) already, so I don't know if the MyXQL adapter should be updated or if this test should be removed.

Reply:
If you can update the adapter to emit the lateral queries that would be great! It should be simply a matter of adding extra clauses here: https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/myxql/connection.ex#L361-L366

A full copy and paste of the PostgreSQL clauses will likely work: https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/postgres/connection.ex#L461-L467

Reply:
Thanks! Created a PR for it https://github.com/elixir-ecto/ecto_sql/pull/339

Reply:
PR merged. Feel free to rebase on top of that but maybe CI will automatically pick it up.

Reply:
I'm guessing CI is failing because the changes from https://github.com/elixir-ecto/ecto/pull/3642 need to be present for this PR. How are PRs that span multiple repositories handled?

Reply:
Did I close that PR by accident? 🤔 If so, what we usually do is this:

1. Change this PR to point to your fork on the mix.exs with `github: ..., override: true, branch: ...`
2. We merge that PR first and we update it to point to Ecto master
3. We merge this

The PR on ecto is failing some unit tests though. Can you please take a look? Thanks!

Reply:
Maybe you closed that PR because you thought it was an issue :man_shrugging:

Okay, I'm doing step 1, I updated the ecto unit tests but looks like I still need to fix some in ecto_sql here.

Reply:
Yeah, sorry about that. Unit tests still failing, can you please take a look? :)

Reply:
All green here now! Can the ecto PR be merged yet?